framework/core/allocated.h

```diff
@@ -173,12 +259,49 @@ class AllocatedBase
 	}
 
   protected:
-	virtual void             post_create(VmaAllocationInfo const &allocation_info);
-	void                     clear();
+	/**
+	 * @brief Internal method to actually create the buffer, allocate the memory and bind them.
+	 * Should only be called from the `Buffer` derived class.
+	 *
+	 * Present in this common base class in order to allow the internal state members to remain `private`
+	 * instead of `protected`, and because it (mostly) isolates interaction with the VMA to a single class
+	 */
 	[[nodiscard]] BufferType create_buffer(BufferCreateInfoType const &create_info);
-	[[nodiscard]] ImageType  create_image(ImageCreateInfoType const &create_info);
-	void                     destroy_buffer(BufferType buffer);
-	void                     destroy_image(ImageType image);
+	/**
+	 * @brief Internal method to actually create the image, allocate the memory and bind them.
+	 * Should only be called from the `Image` derived class.
+	 *
+	 * Present in this common base class in order to allow the internal state members to remain `private`
+	 * instead of `protected`, and because it (mostly) isolates interaction with the VMA to a single class
+	 */
+	[[nodiscard]] ImageType create_image(ImageCreateInfoType const &create_info);
+	/**
+	 * @brief The post_create method is called after the creation of a buffer or image to store the allocation info internally.  Derived classes
+	 * could in theory override this to ensure any post-allocation operations are performed, but the base class should always be called to ensure
+	 * the allocation info is stored.
+	 * Should only be called in the corresping `create_xxx` methods.
+	 */
+	virtual void post_create(VmaAllocationInfo const &allocation_info);
+
+	/**
+	 * @brief Internal method to actually destroy the buffer and release the allocated memory.  Should
+	 * only be called from the `Buffer` derived class.
+	 * Present in this common base class in order to allow the internal state members to remain `private`
+	 * instead of `protected`, and because it (mostly) isolates interaction with the VMA to a single class
+	 */
+	void destroy_buffer(BufferType buffer);
+	/**
+	 * @brief Internal method to actually destroy the image and release the allocated memory.  Should
+	 * only be called from the `Image` derived class.
+	 * Present in this common base class in order to allow the internal state members to remain `private`
+	 * instead of `protected`, and because it (mostly) isolates interaction with the VMA to a single class
+	 */
+	void destroy_image(ImageType image);
+	/**
+	 * @brief Clears the internal state.  Can be overridden by derived classes to perform additional cleanup of members.
```

Comment:
Can it be overridden? It's not marked as virtual.

Reply:
I don't think there's a need for this to be overriden.

Reply:
I don't think so either. So please remove that comment.